configure.ml

```diff
@@ -55,70 +58,70 @@ let opam_var v =
   let _ = Unix.close_process_in ch in
   s
 
+let concat = Filename.concat
```

Comment:
> Having generated files checked into the repository is really a sure way to have them out-of-sync at one time, but theses files have to be part of the repo in order to pin the repo. At the very least, a CI test checking that they are in sync is needed.

I don't understand this part.

> while I'm at it, for any PR, please separate commit with semantics changes from commits which reformats code, especially on large scale, because else the review becomes almost impossible

That's not the case in this PR, no?

Reply:
Forgot about the alias, I used it to test something and forgot to revert it.

Reply:
> > Having generated files checked into the repository is really a sure way to have them out-of-sync at one time, but theses files have to be part of the repo in order to pin the repo. At the very least, a CI test checking that they are in sync is needed.
> 
> I don't understand this part.
> 

When files that are generated by some command, are also committed into the repo, you can be absolutely 100% certain that at some point someone will modify the `dune-project`, commit that change, forget to re-generate the `.opam` files, and push the change to the `dune-project` file without the associated changes that should also occur in the `opam` files, at which point the files are out-of-sync on the repo, and it will introduce strnage changes in unrelated commits... Basically commiting generated file is **BAD** and dangerous in general, and we definitely need to have the .opam files committed currently.

> > while I'm at it, for any PR, please separate commit with semantics changes from commits which reformats code, especially on large scale, because else the review becomes almost impossible
> 
> That's not the case in this PR, no?

Well, in this PR you have at least three changes:
- some moving around of files, and some necessary changes associated to these moves (and these moves **only**)
- some changes to files in order to change some things (these are not strictly related to the move of files)
- some files have been modified because of the ocamlformat reformatting

What I'm saying in that one should make sure in PRs such as this that these three kind of changes in are separate commits in order to make the PR reviewable. Because currently there are hundreds of files modified, most of which should only contain formatting changes (i.e. no semantic changes) which needn't be reviewed with much attention if we trust ocamlformat, but there are also other changes (for instance the changes to the .opam file build intructions) that have to be reviewed carefully. The problem is that in its current state, this is very complicated to do because all changes are thrown in together and it's not easy to spot which change is which....

Reply:
Okay so it seems there are no ocamlformat re-formatting, i just thought so because of all the unnecessary changes to the configure script.
It might be useful to have more accurate and complete description of the changes either in a PR description (or detailed in commit messages), ^^

Reply:
> When files that are generated by some command, are also committed into the repo, you can be absolutely 100% certain that at some point someone will modify the dune-project, commit that change, forget to re-generate the .opam files, and push the change to the dune-project file without the associated changes that should also occur in the opam files, at which point the files are out-of-sync on the repo, and it will introduce strnage changes in unrelated commits... Basically commiting generated file is BAD and dangerous in general, and we definitely need to have the .opam files committed currently.

Ah, yes, I agree. 

No files have been changed by ocamlformat so I disagree with the rest of your message.

Reply:
Yep, I should put a better commit message, as of now it's just a work in progress on which I didn't ask for a review, I just wanted you to notice I started working on it, don't worry, when I'll need you it'll be well documented ;-)

Reply:
Well, currently, the diff of the PR makes it quite hard to see that you actually broke the opam file (the missing configure step), because with the move and the generation of files combined, you don't see, nor usually review the diff of the `.opam` files because it doesn't appear as a diff (you see one file removed and one file added, but not them side by side), and it is easy to miss them when reviewing. Whereas, if the move of files we in a commit of itw own, and the addition of dune-generated files was in another commit of its own, it would be much much much easier to review.

Reply:
I'm pretty sure one much is enough. ;-)
And as I already said, once everything is done I will commit the changes group by group, don't start to review them now, it's just a loss of time ;-)